### ISSUE TITLE
RS1013 claims that RegisterCompilationStartAction + RegisterCompilationEndAction == RegisterCompilationAction, this is incorrect.

### ISSUE THREAD

**Issue by dpaoliello** [dotnet/roslyn-analyzers#1099]
#### Analyzer package

Microsoft.CodeAnalysis.Analyzers

#### Analyzer

StartActionWithOnlyEndAction
Start action has no registered non-end actions.

#### Repro steps

```csharp
[DiagnosticAnalyzer(LanguageNames.CSharp)]
public sealed class Analyzer : DiagnosticAnalyzer
{
    public override void Initialize(AnalysisContext context)
    {
        context.RegisterCompilationStartAction(OnCompilationStart);
    }

    private void OnCompilationStart(CompilationStartAnalysisContext context)
    {
        // RS1013 here:
        context.RegisterCompilationEndAction(OnCompilationEnd);
    }

    private void OnCompilationEnd(CompilationAnalysisContext obj)
    {
    }
}
```

#### Expected behavior
No error.

Per https://github.com/dotnet/roslyn/blob/master/docs/analyzers/Analyzer%20Actions%20Semantics.md the compilation start and end action are guaranteed to be the very first and last actions (respectively), whereas there are no guarantees for the compilation action.

A quick test:
```csharp
[DiagnosticAnalyzer(LanguageNames.CSharp)]
public sealed class Analyzer : DiagnosticAnalyzer
{
    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(new DiagnosticDescriptor("TEST0001", "Test", "Test", "Test", DiagnosticSeverity.Warning, isEnabledByDefault: true));

    public override void Initialize(AnalysisContext context)
    {
        context.RegisterSyntaxTreeAction(OnSyntaxTree);
        context.RegisterCompilationAction(OnCompilation);
        context.RegisterCompilationStartAction(OnCompilationStart);
    }

    private void OnSyntaxTree(SyntaxTreeAnalysisContext obj)
    {
        Console.WriteLine("SyntaxTree");
    }

    private void OnCompilationStart(CompilationStartAnalysisContext context)
    {
        Console.WriteLine("CompilationStartAction");
        context.RegisterCompilationEndAction(OnCompilationEnd);
    }

    private void OnCompilationEnd(CompilationAnalysisContext obj)
    {
        Console.WriteLine("CompilationEndAction");
    }

    private static void OnCompilation(CompilationAnalysisContext context)
    {
        Console.WriteLine("CompilationAction");
    }
}
```

Shows that the `CompilationAction` event tends to happen early (note that these results are unstable):
```
CompilationStartAction
CompilationAction
SyntaxTree
CompilationEndAction
```

#### Actual behavior
RS1013 is raised.

**Comment by mavasani** [comment]
The analyzer fires this diagnostic **only** when you register just a CompilationEndAction on a CompilationStartAction. Given that you have no intermediate registered actions, there is not stateful analysis done by the Start-End action pair. Hence the analyzer correctly tells you to move all the analysis into a single time independent CompilationAction.

**Comment by dpaoliello** [comment]
@mavasani - I'm sorry, but I disagree with this assessment: the analyzer will flag a Start-End pair where all of the analysis is being done in the CompilationEndAction (i.e. the CompilationStartAction only registers the CompilationEndAction). 
It will then tell the user to move their analysis code from CompilationEndAction into CompilationAction - this is incorrect as CompilationEndAction and CompilationAction are NOT equivalent - their timing is different, which may result in bugs if the user follows the recommendation.

**Comment by Denis535** [comment]
@dpaoliello how do you view console output?
